partiql-lang/src/main/kotlin/org/partiql/lang/syntax/impl/PartiQLPigVisitor.kt

```diff
@@ -963,7 +1003,8 @@ internal class PartiQLPigVisitor(val customTypes: List<CustomType> = listOf(), p
      *
      */
 
-    override fun visitExprTermWrappedQuery(ctx: PartiQLParser.ExprTermWrappedQueryContext) = visit(ctx.expr(), PartiqlAst.Expr::class)
+    override fun visitExprTermWrappedQuery(ctx: PartiQLParser.ExprTermWrappedQueryContext) =
+        visitExpr(ctx.expr())
```

Comment:
Here we can avoid a cast (at this call site, at least), by calling a more specialized method (like `visitExpr` here), which already returns a result of the desired type. 
In many cases, this approach is already in place -- see calls to `visitSymbolPrimitive`, `visitPathSimple`, so this is nothing new.